src/plugins/plugin-immer.ts

```diff
@@ -7,15 +7,29 @@
  */
 
 import produce from 'immer';
-import { Plugin } from '../types';
+import { AnyFn, Ctx, Plugin } from '../types';
+import { INVALID_MOVE } from '../core/reducer';
```

Comment:
We should move this constant into it's own file to avoid the circular dependency:

```
(!) Circular dependency: src/plugins/main.ts -> src/plugins/plugin-
immer.ts -> src/core/reducer.ts -> src/plugins/main.ts
(!) Circular dependency: src/core/game.ts -> src/plugins/main.ts ->
 src/plugins/plugin-immer.ts -> src/core/reducer.ts -> src/core/gam
e.ts
```